File: day1/javascript/day1.js

const data = require('./datainput.json');

//part 1
function GetTotalIncreases(inputArray) {
    
    let prevValue = 0;
    let totalIncreases = 0;

    for (let index = 0; index < inputArray.length; index++) {

        if(index !== 0 && prevValue < inputArray[index]) {
            totalIncreases++
        }

        prevValue = inputArray[index]
    }

    return totalIncreases
}

console.log("Part 1 answer: " + GetTotalIncreases(data))

//part 2

function GetSlidingIncreases(inputArray) {
    let prevSum = 0;
    let totalIncreases = 0;

    for (let i = 0; i < inputArray.length; i++) {
        
        let currentSum = GetSums(inputArray[i], inputArray[i + 1], inputArray[i + 2])

        if(i !== 0 && prevSum < currentSum) {
            totalIncreases++
        }
        
        prevSum = currentSum;
    }

    return totalIncreases;
}

function GetSums(val1, val2, val3) {
    return val1 + val2 + val3
}

console.log("Part 2 answer: " + GetSlidingIncreases(data))

File: day1/python/day1.py
import json

jsonFile = open("data.json")

data = json.load(jsonFile)

def GetPart1(input):
    total_increases = 0
    prev_value = 0

    for item in input:
        if prev_value != 0 and prev_value < item:
            total_increases += 1

        prev_value = item

    return total_increases

def GetPart2(input):
    total_increases = 0
    prev_value = 0

    for idx, elem in enumerate(input):
        sliding_sum = 0

        if idx + 2 < len(input):
            sliding_sum = elem + input[idx + 1] + input[idx + 2]
            
        if prev_value != 0 and prev_value < sliding_sum:
            total_increases += 1       

        prev_value = sliding_sum

    return total_increases


print(f"Part 1 answer: {GetPart1(data)}")
print(f"Part 2 answer: {GetPart2(data)}")

